paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentControllerUnitTests.java

```diff
@@ -422,14 +496,15 @@ public void whenCheckReturnUrlWithEmptySettlementThenReturnStatus200() throws Ex
 
         TreeMap<String, String> filteredParams = PaytrailSignatureService.filterCheckoutQueryParametersMap(mockCallbackCheckoutParams);
         try {
-            String mockSignature = PaytrailSignatureService.calculateSignature(filteredParams, null, secretKey);
+            String mockSignature = PaytrailSignatureService.calculateSignature(filteredParams, null, PAYTRAIL_SECRET_KEY);
             mockCallbackCheckoutParams.put("signature", mockSignature);
 
             /* Convert mock callback checkout params to actual query params */
             String queryParams = mockCallbackCheckoutParams.keySet().stream()
                     .map(key -> key + "=" + mockCallbackCheckoutParams.get(key))
                     .collect(Collectors.joining("&"));
-
+            queryParams += "&" + "merchantId=" + TEST_MERCHANT_ID;
```

Comment:
you could just add this to mockCallbackCheckoutParams before? 
`mockCallbackCheckoutParams.put("merchantId", TEST_MERCHANT_ID)`